### PR TITLE
[Cleanup] Number Precision And Decimal Separator Adjusting

### LINKED ISSUE #2057: Wrong duration formatting in invoice with task details
## Setup
- Version: v5.10.27
- Environment: Shared Hosting

## Interface
- Flutter: []
- React: [X]
- Both: []

## Checklist
- Can you replicate the issue on our v5 demo site https://demo.invoiceninja.com or **https://react.invoicing.co/demo?**
React Demo behaves similar to self hosted, Flutter drops errors when account+client country set to "german": Invalid country **ID.**
- Have you searched existing issues?
Yes, Forum
- Have you reported this to Slack/forum before posting?
No
- Have you inspected the logs in storage/logs/laravel.log for any errors?
No hint in logs.

## Describe the bug
Setting localisation like this:
Bildschirmfoto 2024-09-17 um 11 46 02
Task settings like this:
Bildschirmfoto 2024-09-17 um 11 46 58
Results in Invoice-Details like this (should be "2,25 Stunden" instead of "2.2500 Stunden") wrong decimal-separator, too many decimals:
Bildschirmfoto 2024-09-17 um 11 51 43
Bildschirmfoto 2024-09-17 um 12 07 59

### Steps To Reproduce
setup like screenshots above

### Expected Behavior
Formatting should be e.g "2,25 Stunden" instead of "2.2500 Stunden"

## Additional context
The react backend has no option to influence this behaviour probably (see flutter backend):
Bildschirmfoto 2024-09-17 um 12 50 39
The Linux App of which I guess is flutter-based works also not as expected – and not stable enough (Ubuntu 24.04.1 LTS, IN 5.0.158). It delivers mixed results:
Bildschirmfoto vom 2024-09-17 13-26-41
 

### Screenshots
-above-

### Logs
-don't contribute to the problem.

### PR DESCRIPTION
**Pull request by Civolilah**
@beganovich @turbo124 The PR includes adjustments for decimal precision and separators on various functionalities such as invoicing products, projects, tasks, etc. The value used for decimal precision will be the one entered in `User Details => Preferences`. If no value is entered, the default precision will be `2`. Another part of the adjustment concerns separators. For example, when creating notes while invoicing a project, we will use the separator as defined in the `Localization` settings. Let me know your thoughts.

Closes #2057

### CODE REVIEW

**Comment by Civolilah** [comment]
@turbo124 A separate PR will be created for fixes on the Preview table in Reports. We should make corrections there for monetary value displays.

**Comment by Civolilah** [comment]
@turbo124 The adjustment has been made here, so formatting numbers on values such as descriptions, where it is something that the API cannot format but is also important to send to the client's customer, is formatted at the client level. However, when the formatting is not related to the client's customer, we will format values according to company settings. Let me know your thoughts.

**Comment by turbo124** [comment]
image

When are you format the time in the task log (see image) it must be localized to the client. note here this client is DEU + EUR, so comma separator is used for currency, but not for time duration.

**Comment by Civolilah** [comment]
> image
> 
> When are you format the time in the task log (see image) it must be localized to the client. note here this client is DEU + EUR, so comma separator is used for currency, but not for time duration.

@turbo124 Ah, you are right! I missed it. So right now, the zones, date, and time formats should be adjusted on invoicing projects, invoicing tasks, and adding tasks to an invoice. Let me know your thoughts.

**Comment by Civolilah** [comment]
@turbo124 As we discussed on Slack, the offset addition has been removed here. Only formatting should be applied for all date/time values, along with number separator and precision formatting. Let me know your thoughts.

File: src/common/hooks/useNumericFormatter.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        : company?.use_comma_as_decimal_place
+        ? '.'
+        : ',',
+      decimalSeparator: decimalSeparator
+        ? decimalSeparator
+        : company?.use_comma_as_decimal_place
+        ? ','
+        : '.',
+      decimalScale: precision || userNumberPrecision,
+    });
```
Comment (beganovich) [inline src/common/hooks/useNumericFormatter.ts:37]: Please refactor with nice if statements, this is really unreadable with so many nesting levels.

Comment (Civolilah) [reply src/common/hooks/useNumericFormatter.ts:37]: @beganovich Sure thing, the logic has been separated into functions and then applied to the logic. Let me know your thoughts.

**Review by beganovich — APPROVED** [review]

**Pull request merged**